### PR TITLE
Add MountEITBasedFileShare to mounter interface

### PR DESCRIPTION
**Pull request by prankulmahajan**

### CODE REVIEW

File: pkg/mountmanager/mount_linux.go [hunk 1]
```diff
@@ -75,3 +115,54 @@ func (m *NodeMounter) Resize(devicePath string, deviceMountPath string) (bool, e
...
+		socketPath = defaultSocketPath
+	}
+	// Create a custom dialer function for Unix socket connection
+	dialer := func(ctx context.Context, network, addr string) (net.Conn, error) {
+		return net.Dial("unix", socketPath)
+	}
+
+	// Create an HTTP client with the Unix socket transport
+	client := &http.Client{
+		Transport: &http.Transport{
```
Comment (prankulmahajan) [inline pkg/mountmanager/mount_linux.go:133]: Put timeout of 3 min

File: pkg/mountmanager/mount_linux.go [hunk 2]
```diff
@@ -75,3 +115,54 @@ func (m *NodeMounter) Resize(devicePath string, deviceMountPath string) (bool, e
...
+		Error         string `json:"Error"`
+		ErrorResponse string `json:"Response"`
+	}
+	err = json.Unmarshal(body, &responseBody)
+	if err != nil {
+		return "", err
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return responseBody.ErrorResponse, fmt.Errorf("Response from mount-helper-container -> Exit Status Code: %s ,ResponseCode: %v", responseBody.Error, response.StatusCode)
```
Comment (prankulmahajan) [inline pkg/mountmanager/mount_linux.go:165]: (GoodToHave) Put in common place

**Review by sameshai — APPROVED** [review]

**Review by sameshai — APPROVED** [review]

File: pkg/mountmanager/mount_linux.go [hunk 3]
```diff
@@ -75,3 +104,55 @@ func (m *NodeMounter) Resize(devicePath string, deviceMountPath string) (bool, e
...
+	dialer := func(ctx context.Context, network, addr string) (net.Conn, error) {
+		return net.Dial("unix", socketPath)
+	}
+
+	// Create an HTTP client with the Unix socket transport
+	client := &http.Client{
+		Transport: &http.Transport{
+			DialContext: dialer,
+		},
+		Timeout: 3 * time.Minute, // Put 3 min timeout
```
Comment (arahamad) [inline pkg/mountmanager/mount_linux.go:125]: This 3 mins should be configurable or const not like this hard coded value

**Review by arahamad — APPROVED** [review]
lgtm, please verify also

File: pkg/mountmanager/mount_linux.go (hunk 3 above)
Comment (prankulmahajan) [reply pkg/mountmanager/mount_linux.go]: Moved as a const

**Pull request merged**